app/models/ambassador_task.rb

```diff
@@ -14,14 +14,14 @@ def description_html
 
   # Assign the receiver to the given Ambassador
   # Return the AmbassadorTaskAssignment instance
-  def assign_to(user)
-    ambassador_task_assignments.create(user: user)
+  def assign_to(ambassador)
+    ambassador_task_assignments.create(ambassador: ambassador)
   end
 
   # Assigns the task to all ambassadors, if not already assigned
   def ensure_assigned_to_all_ambassadors!
     Ambassador.find_each do |ambassador|
-      next if ambassador_task_assignments.exists?(user: ambassador)
+      next if ambassador_task_assignments.exists?(ambassador: ambassador)
```

Comment:
Does this instantiate every ambassador? aka, will this generate performance issues and should this use pluck & array comparison outside of the `find_each`, and batch generate the tasks - for when we have 100 ambassadors?

Reply:
We could definitely do away with instantiating models but most of the time is spent in the db — for 100 users this'll take around 15 seconds (extrapolating based on the 29 ambassador records I have locally). I think it would be a good idea to (a) move this logic into a background job or (b) bulk insert